### ISSUE TITLE
Delay Screen Navigation when using TextInput

### ISSUE THREAD

**Issue by saadi-ninjasCode** [callstack/react-native-paper#3601]
### Current behaviour
When navigating **screen B** from **Screen A**. **Screen B** has some text inputs (even single input). When pressing a button on  **Screen A**, **Screen A** gets stuck for some seconds (e.g 2 sec). Screen navigation starts after some seconds(e.g 2 sec)
When I experiment with React Native text input, I didn't face such delays.

This behaviour can be seen in react-native-paper [snack example ](https://snack.expo.dev/@react-native-paper/react-native-paper-example_v5)

### Expected behaviour
There will be no such delay in screen navigation.

### How to reproduce?
This behaviour can be seen in react-native-paper [snack example ](https://snack.expo.dev/@react-native-paper/react-native-paper-example_v5)

### Your Environment

| software                      | version
| --------------------- | -------
| react-native                | 0.71.0
| react-native-paper    | 5.1.3
| node                            | 18.13.0
|  yarn                | 1.22.18

**Comment by lukewalczak** [comment]
Hey @saadi-ninjasCode, could you please extract the specific code to the separated snack?

**Comment by saadi-ninjasCode** [comment]
> Hey @saadi-ninjasCode, could you please extract the specific code to the separated snack?

Here is the link of the separated Snack.
[React Native Paper Text Input](https://snack.expo.dev/@saad-ninjascore/react-native-paper-textinput)
 Furthermore, I am also attaching a video in the description.

**Comment by lukewalczak** [comment]
I think the screen with TextInput's is overloaded and needs some optimization. Once I left there only eg 4 of them, there is no delay. Do you experience the similar issue in your project or just on the example snack?

**Comment by saadi-ninjasCode** [comment]
> I think the screen with TextInput's is overloaded and needs some optimization. Once I left there only eg 4 of them, there is no delay. Do you experience the similar issue in your project or just on the example snack?

I experience a similar issue in my project. I have 6 inputs on my screen. Then I run the React Native Snack example and I face the same issue there. That's why I report this issue.

When I use the Bare React Native TextInput I didn't face such an issue.

**Comment by RichardLindhout** [comment]
The textinput is currently really slow indeed!

**Comment by lukewalczak** [comment]
Hey @saadi-ninjasCode, could you please check whether the delay is visible using the latest version 5.2.0?

**Comment by saadi-ninjasCode** [comment]
> Hey @saadi-ninjasCode, could you please check whether the delay is visible using the latest version 5.2.0?

I didn't notice any improvement in delay.

**Comment by EvgeniyaGorobets** [comment]
I'm facing a similar problem. I've got a recipe app and the `RecipeForm` screen has quite a few TextInput elements. On my Android phone (Google Pixel 4), there's a huge delay when I navigate to this screen. I tried to use React DevTools to profile my app, but it shows that the initial render of the recipe form takes 71.7ms:

<img width="1727" alt="image" src="https://github.com/callstack/react-native-paper/assets/54817713/56c89297-6632-46b7-ba33-8b978465fc00">

If I hover over any of the components in the profiler (in the image above), it lists the reason for render as "This is the first time this component rendered". Even if I optimize my app to get rid of rerenders, the first render is so slow that the app feels laggy.

I'm on v5.10.3 of react-native-paper, v0.71.1 of react-native, v47.0.0 of expo, and I'm using v7.47.0 of react-hook-form to optimize my form.

**Comment by praweshlamsal** [comment]
> I'm facing a similar problem. I've got a recipe app and the `RecipeForm` screen has quite a few TextInput elements. On my Android phone (Google Pixel 4), there's a huge delay when I navigate to this screen. I tried to use React DevTools to profile my app, but it shows that the initial render of the recipe form takes 71.7ms:
> 
> <img alt="image" width="1727" src="https://private-user-images.githubusercontent.com/54817713/273648918-56c89297-6632-46b7-ba33-8b978465fc00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcyMDExNzcsIm5iZiI6MTcwNzIwMDg3NywicGF0aCI6Ii81NDgxNzcxMy8yNzM2NDg5MTgtNTZjODkyOTctNjYzMi00NmI3LWJhMzMtOGI5Nzg0NjVmYzAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA2VDA2Mjc1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2YzA2NGExM2NhZTBjNzhhY2QwNDk3ZjEyMTVmY2NkYTkxOWE3NTU4ZTE5MjA0MDUyMGRmY2I4OGIyZmI0YjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yP2hllHt4Ab-K2NUqTWPoY9AdA2hMPS_s635O-BZvBo">
> If I hover over any of the components in the profiler (in the image above), it lists the reason for render as "This is the first time this component rendered". Even if I optimize my app to get rid of rerenders, the first render is so slow that the app feels laggy.
> 
> I'm on v5.10.3 of react-native-paper, v0.71.1 of react-native, v47.0.0 of expo, and I'm using v7.47.0 of react-hook-form to optimize my form.

Is there any solution for this ? Have you got one ?

**Comment by EvgeniyaGorobets** [comment]
@praweshlamsal No, unfortunately. This was for a personal project so I just kept it as is 🤷‍♀️

**Comment by mrakus-sofomo** [comment]
I got this issue when I used `react-native-paper` together with `react-hook-form`. Bare inputs, without form validator, works fine.

**Comment by abhishek-QSPL0071** [comment]
Any workaround or fix for this?
